lib/common/versionUtils.js

```diff
@@ -129,7 +130,32 @@ function getConcreteSchemaUtils({ type, data }) {
   return concreteUtils;
 }
 
+/**
+   * Filter the options by the version provided by the user
+   * @param {array} optsArray The options to be filtered
+   * @param {string} version The version that will be used
+   * @returns {array} the filtered options related to the version used
+   */
+function filterOptionsByVersion(optsArray, version) {
+  let defOptions = {};
+  optsArray = optsArray.filter((option) => {
+    return option.supportedIn.includes(version);
+  });
+  _.each(optsArray, (opt) => {
+    // special handling for indent character as in documentation it states `Tab` and `Space`
+    // but for the generation mode, we need actual values
+    if (opt.id === 'indentCharacter') {
```

Comment:
i don't think that this business rule should be here, maybe we should just let the filter code